src/main/webapp/profile.js

```diff
@@ -13,15 +13,37 @@
 // limitations under the License.
 
 function onProfileLoad() {
+  autofillForm(getUserIfRegistered());  
+  supplyLogoutLink();
   prepareFormValidation();
 }
 
-// Allows certain fields in the profile to be edited, hides edit button, and displays update button. 
+// Returns a Person if they registered in the past. If not, redirect to  
+// registration page.
+function getUserIfRegistered(){
+  fetch('/login')
+  .then(response => response.json())
+  .then(status => fetch(`/person?email=${status.email}`))
+  .then(response => {
+    if (response.redirected) {
+      window.location.href = response.url;
+      return;
+    }
+    return response.json();
+  });
+}
 
-function onProfileLoad() {
-  supplyLogoutLink();
+// Fills in the profile form with data from Datastore.
+function autofillForm(person) {
+  document.getElementById("user-email").value = person.email;
+  document.getElementById("first-name-field").value = person.firstName;
+  document.getElementById("last-name-field").value = person.lastName;
+  document.getElementById("company-field").value = person.company;
+  document.getElementById("job-field").value = person.job;
+  document.getElementById("linkedin-field").value = person.linkedIn;    
```

Comment:
' is preferred over " https://sites.google.com/a/google.com/dfa6/development/javascript-1/JavaScript-Style-Guide